haystack_experimental/components/agents/agent.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import inspect
+from typing import Any, Optional, Union
+
+from haystack import logging
+from haystack.components.agents.agent import Agent as HaystackAgent
+from haystack.components.agents.state.state import _validate_schema
+from haystack.components.agents.state.state_utils import merge_lists
+from haystack.components.generators.chat.types import ChatGenerator
+from haystack.core.component.component import component
+from haystack.core.pipeline.utils import _deepcopy_with_exceptions
+from haystack.dataclasses import ChatMessage
+from haystack.dataclasses.streaming_chunk import StreamingCallbackT
+from haystack.tools import Tool, Toolset
+
+from haystack_experimental.components.tools.tool_invoker import ToolInvoker
+
+logger = logging.getLogger(__name__)
+
+
+class Agent(HaystackAgent):
+    def __init__(
+        self,
+        *,
+        chat_generator: ChatGenerator,
+        tools: Optional[Union[list[Tool], Toolset]] = None,
+        system_prompt: Optional[str] = None,
+        exit_conditions: Optional[list[str]] = None,
+        state_schema: Optional[dict[str, Any]] = None,
+        max_agent_steps: int = 100,
+        streaming_callback: Optional[StreamingCallbackT] = None,
+        raise_on_tool_invocation_failure: bool = False,
+        tool_invoker_kwargs: Optional[dict[str, Any]] = None,
```

Comment:
Even if I understand that we are not changing the `__init__`, as a user, I would find more
straightforward to set `confirmation_strategies` in the `Agent`, not in `tool_invoker_kwargs`.